Userland/Applications/SoundPlayer/main.cpp

```diff
@@ -19,118 +20,113 @@
 #include <LibGUI/Menubar.h>
 #include <LibGUI/Window.h>
 #include <LibGfx/CharacterBitmap.h>
+#include <LibMain/Main.h>
 #include <stdio.h>
 
-int main(int argc, char** argv)
+ErrorOr<int> serenity_main(Main::Arguments arguments)
 {
-    if (pledge("stdio recvfd sendfd rpath thread unix", nullptr) < 0) {
-        perror("pledge");
-        return 1;
-    }
+    TRY(Core::System::pledge("stdio recvfd sendfd rpath thread unix"));
 
-    auto app = GUI::Application::construct(argc, argv);
-    auto audio_client = Audio::ClientConnection::construct();
+    auto app = TRY(GUI::Application::try_create(arguments));
+    auto audio_client = TRY(Audio::ClientConnection::try_create());
 
-    if (pledge("stdio recvfd sendfd rpath thread", nullptr) < 0) {
-        perror("pledge");
-        return 1;
-    }
+    TRY(Core::System::pledge("stdio recvfd sendfd rpath thread"));
 
     auto app_icon = GUI::Icon::default_icon("app-sound-player");
 
-    auto window = GUI::Window::construct();
+    auto window = TRY(GUI::Window::try_create());
     window->set_title("Sound Player");
     window->set_icon(app_icon.bitmap_for_size(16));
 
-    String path = argv[1];
+    String path = arguments.argv[1];
     // start in advanced view by default
-    Player* player = &window->set_main_widget<SoundPlayerWidgetAdvancedView>(window, audio_client);
-    if (argc > 1) {
+    Player* player = TRY(window->try_set_main_widget<SoundPlayerWidgetAdvancedView>(window, audio_client));
+    if (arguments.argc > 1) {
```

Comment:
Accessing the argument before ensuring it exists is a silent index-out-of-bounds and could be easily drive-by-fixed:
```suggestion
    // start in advanced view by default
    Player* player = TRY(window->try_set_main_widget<SoundPlayerWidgetAdvancedView>(window, audio_client));
    if (arguments.argc > 1) {
        String path = arguments.strings[1];
```

Reply:
Thanks, nice catch! 🐞